uint32 start_x
uint32 start_y
uint32 width
uint32 height